jmix-flowui/flowui/src/main/java/io/jmix/flowui/data/binding/impl/HtmlContainerReadonlyDataBindingImpl.java

```diff
@@ -0,0 +1,172 @@
+
+        Subscription itemChangeSubscription = dataContainer.addItemChangeListener(itemChangeEvent -> {
+            Object propertyValue = itemChangeEvent.getItem() != null
+                    ? EntitySystemAccess.getEntityEntry(itemChangeEvent.getItem()).getAttributeValue(property)
+                    : null;
+            updateHtmlContainerText(htmlContainer, formatter, propertyValue);
+        });
+        HtmlContainerListenerRegistrationsCleaner itemChangeCleaner = itemChangeSubscription::remove;
+        listenersRegistrations.add(new Pair<>(htmlContainer, itemChangeCleaner));
+    }
+
+    @Override
+    public void bind(HtmlContainer htmlComponent, CollectionContainer<?> dataContainer) {
+        bind(htmlComponent, dataContainer, new HtmlContainerTextFormatter());
+    }
+
+    @Override
+    public void bind(HtmlContainer htmlContainer, CollectionContainer<?> dataContainer, Formatter<Object> formatter) {
+        Preconditions.checkNotNullArgument(htmlContainer);
+        Preconditions.checkNotNullArgument(dataContainer);
+        Preconditions.checkNotNullArgument(formatter);
+
+        updateHtmlContainerText(htmlContainer, formatter, dataContainer.getItems());
+
+        Subscription subscription = dataContainer.addCollectionChangeListener(collectionChangeEvent ->
+                updateHtmlContainerText(htmlContainer, formatter, dataContainer.getItems()));
+
+        HtmlContainerListenerRegistrationsCleaner htmlContainerListenerRegistrationsCleaner = subscription::remove;
+        listenersRegistrations.add(new Pair<>(htmlContainer, htmlContainerListenerRegistrationsCleaner));
+    }
+
+    public void unbind(HtmlContainer htmlContainer) {
+        Preconditions.checkNotNullArgument(htmlContainer);
+
+        Iterator<Pair<HtmlContainer, HtmlContainerListenerRegistrationsCleaner>> iterator = listenersRegistrations.iterator();
+        while (iterator.hasNext()) {
+            Pair<HtmlContainer, HtmlContainerListenerRegistrationsCleaner> htmlContainerAndCleaner = iterator.next();
+            if (htmlContainer.equals(htmlContainerAndCleaner.getFirst())) {
+                htmlContainerAndCleaner.getSecond().remove();
+                iterator.remove();
+            }
+        }
+    }
+
+    protected void updateHtmlContainerText(HtmlContainer htmlContainer,
+                                           Formatter<Object> formatter,
+                                           @Nullable Object value) {
+        htmlContainer.setText(formatter.apply(value));
+    }
+
+    @FunctionalInterface
+    protected interface HtmlContainerListenerRegistrationsCleaner {
+        void remove();
+    }
+
+    class HtmlContainerTextFormatter implements Formatter<Object> {
+
+        @Override
+        public String apply(@Nullable Object value) {
+            if (value instanceof Collection<?>) {
+                return formatCollection((Collection<?>) value);
+            }
```

Comment:
![image](https://github.com/jmix-framework/jmix/assets/75516667/5f681a02-9f30-4ea8-9bf0-2cb2483dc6e5)


Java 17 baby:
```suggestion
            if (value instanceof Collection<?> collectionValue) {
                return formatCollection(collectionValue);
            }
```